scripts/gen-workflow-mermaid.ts

```diff
@@ -0,0 +1,611 @@
+#!/usr/bin/env -S npx ts-node
+
+import fs from "node:fs";
+import path from "node:path";
+import YAML from "yaml";
+import parseArgs from "minimist";
+import cronstrue from "cronstrue";
+import { partition } from "lodash";
+
+const argv = parseArgs<{
+    debug: boolean;
+    on: string | string[];
+}>(process.argv.slice(2), {
+    string: ["on"],
+    boolean: ["debug"],
+});
+
+class IdGenerator {
```

Comment:
this seems to be more than an `IdGenerator`, it also keeps track of a string and its associated id? 

Not sure if there's another name that would represent that better?

Reply:
A better name may be available but I'd argue it is just an ID generator, you give it strings and it generates (in this implementation's case incremental base36) IDs to represent those objects (strings)

Reply:
It makes more sense if the input is generic rather than just string